From f7b03d10b504b1172c3fc6dab33534babf79d035 Mon Sep 17 00:00:00 2001
From: Ganapathi Bhat <gbhat@marvell.com>
Date: Thu, 27 Apr 2017 23:48:10 -0700
Subject: [PATCH 08/13] CHROMIUM: arm64: dts: rockchip: add wifi tx power table
 for bob

This patch adds wifi tx power table for the mwifiex driver to support
dynamic wifi power switching.

BUG=b:35583804
TEST=With this patch series applied, verified on bob that correct data
is downloaded to firmware for below commands.
1) Set power limits to 2.4G high + 5G high.
       ./iw dev mlan0 mwifiex dev_tx_power 1 1
2) Set power limits 2.4G low + 5G low
       ./iw dev mlan0 mwifiex dev_tx_power 0 0

Change-Id: I6eb4d117f5383ee9ddf06698ec9002f634a25b13
Signed-off-by: Ganapathi Bhat <gbhat@marvell.com>
Signed-off-by: Amitkumar Karwar <akarwar@marvell.com>
Reviewed-on: https://chromium-review.googlesource.com/489816
Commit-Ready: Shasha Zhao <Sarah_Zhao@asus.com>
Tested-by: Shasha Zhao <Sarah_Zhao@asus.com>
Reviewed-by: Wei-Ning Huang <wnhuang@chromium.org>
---
 .../boot/dts/rockchip/rk3399-gru-bob.dtsi     | 424 ++++++++++++++++++
 1 file changed, 424 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-gru-bob.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-gru-bob.dtsi
index 8113f436918e..fc5b67808bce 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-gru-bob.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-gru-bob.dtsi
@@ -143,6 +143,430 @@
 	temperature = <65000>;
 };
 
+&mvl_wifi {
+	marvell,caldata_00_txpwrlimit_2g_cfg_set = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x30 0x02 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x01 0x00 0x0d
+0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d 0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d
+0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d 0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x02 0x00 0x0d 0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d
+0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d 0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d
+0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x03 0x00 0x0d
+0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d 0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d
+0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d 0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x04 0x00 0x0d 0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d
+0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d 0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d
+0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x05 0x00 0x0d
+0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d 0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d
+0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d 0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x06 0x00 0x0d 0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d
+0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d 0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d
+0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x07 0x00 0x0d
+0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d 0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d
+0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d 0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x08 0x00 0x0d 0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d
+0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d 0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d
+0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x09 0x00 0x0d
+0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d 0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d
+0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d 0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x0a 0x00 0x0d 0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d
+0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d 0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d
+0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x0b 0x00 0x0d
+0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d 0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d
+0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d 0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x0c 0x00 0x0d 0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d
+0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d 0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d
+0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x0d 0x00 0x0d
+0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d 0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d
+0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d 0x0d 0x0d 0x0e 0x0d 0x0f 0x0d 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x0e 0x00 0x0d 0x01 0x0d 0x02 0x0d 0x03 0x0d 0x04 0x0d
+0x05 0x0d 0x06 0x0d 0x07 0x0d 0x08 0x0d 0x09 0x0d 0x0a 0x0d 0x0b 0x0d 0x0c 0x0d
+0x0d 0x0d 0x0e 0x0d 0x0f 0x0d
+	>;
+
+	marvell,caldata_00_txpwrlimit_5g_cfg_set_sub0 = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x04 0x02 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x24 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x28 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x2c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x30 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x34 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x38 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x3c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x40 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a
+	>;
+
+	marvell,caldata_00_txpwrlimit_5g_cfg_set_sub1 = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x04 0x03 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x64 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x68 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x6c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x70 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x74 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x78 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x7c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x80 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x84 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x88 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x8c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x90 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a
+	>;
+
+	marvell,caldata_00_txpwrlimit_5g_cfg_set_sub2 = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x44 0x01 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x95 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x99 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x9d 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0xa1 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0xa5 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a
+	>;
+
+	marvell,caldata_00_txpwrlimit_5g_cfg_set_sub3 = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x84 0x03 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xb7 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xb8 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xb9 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xbb 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xbc 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xbd 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xc0 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xc4 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x07 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x08 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x0b 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x0c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x10 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x22 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a
+	>;
+
+	marvell,caldata_01_txpwrlimit_2g_cfg_set = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x30 0x02 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x01 0x00 0x0f
+0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f 0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f
+0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f 0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x02 0x00 0x0f 0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f
+0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f 0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f
+0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x03 0x00 0x0f
+0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f 0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f
+0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f 0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x04 0x00 0x0f 0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f
+0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f 0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f
+0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x05 0x00 0x0f
+0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f 0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f
+0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f 0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x06 0x00 0x0f 0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f
+0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f 0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f
+0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x07 0x00 0x0f
+0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f 0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f
+0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f 0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x08 0x00 0x0f 0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f
+0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f 0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f
+0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x09 0x00 0x0f
+0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f 0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f
+0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f 0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x0a 0x00 0x0f 0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f
+0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f 0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f
+0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x0b 0x00 0x0f
+0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f 0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f
+0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f 0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x0c 0x00 0x0f 0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f
+0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f 0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f
+0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01 0x24 0x00 0x67 0x09 0x14 0x0d 0x00 0x0f
+0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f 0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f
+0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f 0x0d 0x0f 0x0e 0x0f 0x0f 0x0f 0x89 0x01
+0x24 0x00 0x67 0x09 0x14 0x0e 0x00 0x0f 0x01 0x0f 0x02 0x0f 0x03 0x0f 0x04 0x0f
+0x05 0x0f 0x06 0x0f 0x07 0x0f 0x08 0x0f 0x09 0x0f 0x0a 0x0f 0x0b 0x0f 0x0c 0x0f
+0x0d 0x0f 0x0e 0x0f 0x0f 0x0f
+	>;
+
+	marvell,caldata_01_txpwrlimit_5g_cfg_set_sub0 = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x04 0x02 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x24 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x28 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x2c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x30 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x34 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x38 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x3c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x40 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a
+	>;
+
+	marvell,caldata_01_txpwrlimit_5g_cfg_set_sub1 = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x04 0x03 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x64 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x68 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x6c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x70 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x74 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x78 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x7c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x80 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x84 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x88 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x8c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x90 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a
+	>;
+
+	marvell,caldata_01_txpwrlimit_5g_cfg_set_sub2 = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x44 0x01 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x95 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x99 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x9d 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0xa1 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0xa5 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a
+	>;
+
+	marvell,caldata_01_txpwrlimit_5g_cfg_set_sub3 = /bits/ 8 <
+0x01 0x00 0x06 0x00 0x84 0x03 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xb7 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xb8 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xb9 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xbb 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xbc 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xbd 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xc0 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0xa0 0x0f 0x14 0xc4 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x07 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x08 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x0b 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x0c 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x10 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a 0x89 0x01 0x3c 0x00 0x88 0x13 0x14 0x22 0x00 0x0a
+0x01 0x0a 0x02 0x0a 0x03 0x0a 0x04 0x0a 0x05 0x0a 0x06 0x0a 0x07 0x0a 0x08 0x0a
+0x09 0x0a 0x0a 0x0a 0x0b 0x0a 0x0c 0x0a 0x0d 0x0a 0x0e 0x0a 0x0f 0x0a 0x10 0x0a
+0x11 0x0a 0x12 0x0a 0x13 0x0a 0x14 0x0a 0x15 0x0a 0x16 0x0a 0x17 0x0a 0x18 0x0a
+0x19 0x0a 0x1a 0x0a 0x1b 0x0a
+	>;
+};
+
 &pinctrl {
 	wifi {
 		wlan_host_wake_l: wlan-host-wake-l {
-- 
2.17.0

